### ISSUE TITLE
Kernel trace when plug-off the device.

### ISSUE THREAD

**Issue by Sauerlandlinux** [lwfinger/rtl8852au#55]
Hi

First of all thanks for the driver.

I bought an BrosTrend AX1800
`Bus 001 Device 035: ID 0bda:8832 Realtek Semiconductor Corp. 802.11ac WLAN Adapter`

A few patching and the driver builds without errors or warnings.
https://build.opensuse.org/package/show/home:Sauerland/rtl8852au

Working as expected but when I pull off the Wlan Device I get:
Aug 23 17:05:30 linux64 kernel: usb 1-1: USB disconnect, device number 23
Aug 23 17:05:30 linux64 kernel: ------------[ cut here ]------------
Aug 23 17:05:30 linux64 kernel: WARNING: CPU: 3 PID: 28234 at /home/abuild/rpmbuild/BUILD/rtl8852au-1.15.0.1~3g716e525/obj/default/phl/phl_role.c:1006 rtw_phl_wifi_role_free+0x27/0x4c [8852au]
Aug 23 17:05:30 linux64 kernel: Modules linked in: 8852au(ON) xfs rfcomm b43legacy(N) b43(N) cordic bcma(N) mac80211 libarc4 ssb mmc_core tun nls_utf8 cifs cifs_arc4 cifs_md4 dns_resolver fscache netfs libdes nft_fib_inet nft_fib_ipv4 nft_fib_ipv6 nft_fib nft_reject_inet nf_reject_ipv4 nf_reject_ipv6 nft_reject nft_ct nft_chain_nat af_packet cmac algif_hash algif_skcipher af_alg nf_tables ebtable_nat ebtable_broute ip6table_nat ip6table_mangle ip6table_raw ip6table_security iptable_nat nf_nat nf_conntrack nf_defrag_ipv6 nf_defrag_ipv4 iptable_mangle iptable_raw iptable_security bridge stp llc iscsi_ibft iscsi_boot_sysfs ip_set nfnetlink ebtable_filter ebtables ip6table_filter ip6_tables iptable_filter vboxnetadp(ON) bpfilter vboxnetflt(ON) vboxdrv(ON) qrtr(N) ns(N) r8153_ecm(N) bnep joydev dmi_sysfs r8152(OX) btusb btrtl btbcm btintel btmtk bluetooth uvcvideo videobuf2_vmalloc videobuf2_memops uas snd_usb_audio videobuf2_v4l2 cdc_ether usb_storage videobuf2_common usbnet snd_usbmidi_lib mii
Aug 23 17:05:30 linux64 kernel:  ecdh_generic videodev snd_rawmidi snd_seq_device mc hid_generic usbhid binfmt_misc intel_rapl_msr intel_rapl_common intel_tcc_cooling(N) nls_iso8859_1 snd_hda_codec_realtek ext4 nls_cp437 x86_pkg_temp_thermal intel_powerclamp snd_hda_codec_generic ledtrig_audio snd_hda_codec_hdmi vfat iTCO_wdt fat intel_pmc_bxt snd_hda_intel coretemp iTCO_vendor_support crc16 mbcache mfd_core mei_hdcp(N) mei_pxp snd_intel_dspcfg ee1004(N) jbd2 snd_intel_sdw_acpi snd_hda_codec kvm_intel eeepc_wmi(N) asus_wmi snd_hda_core battery snd_hwdep mei_me kvm snd_pcm i2c_i801 sparse_keymap snd_timer irqbypass platform_profile wmi_bmof pcspkr efi_pstore(N) i2c_smbus mei snd thermal soundcore fan intel_pmc_core acpi_pad fuse loop configfs ip_tables x_tables sd_mod crc32_pclmul t10_pi sr_mod crc64_rocksoft_generic cdrom crc64_rocksoft crc64 ghash_clmulni_intel wl(PON) r8168(ON) nouveau(N) drm_ttm_helper ttm i2c_algo_bit drm_display_helper mxm_wmi(N) aesni_intel crypto_simd drm_kms_helper
Aug 23 17:05:30 linux64 kernel:  cfg80211 xhci_pci xhci_pci_renesas syscopyarea sysfillrect xhci_hcd sysimgblt fb_sys_fops ahci drm libahci cryptd libata usbcore cec rfkill rc_core wmi video button btrfs blake2b_generic libcrc32c crc32c_intel xor raid6_pq sg dm_multipath dm_mod scsi_dh_rdac scsi_dh_emc scsi_dh_alua scsi_mod msr efivarfs [last unloaded: 8852bu]
Aug 23 17:05:30 linux64 kernel: Supported: No, Proprietary and Unsupported modules are loaded
Aug 23 17:05:30 linux64 kernel: CPU: 3 PID: 28234 Comm: kworker/3:0 Tainted: P        W  O   X  N 5.14.21-150500.55.19-default #1 SLE15-SP5 a29285bac85e8baa0757c512c91f36229b4faaae
Aug 23 17:05:30 linux64 kernel: Hardware name: System manufacturer System Product Name/PRIME Z270-P, BIOS 1205 05/11/2018
Aug 23 17:05:30 linux64 kernel: Workqueue: usb_hub_wq hub_event [usbcore]
Aug 23 17:05:30 linux64 kernel: RIP: 0010:rtw_phl_wifi_role_free+0x27/0x4c [8852au]
Aug 23 17:05:30 linux64 kernel: Code: cc cc cc 0f 1f 44 00 00 40 84 f6 74 05 c3 cc cc cc cc 55 53 48 89 fb 48 8b af 18 11 00 00 48 89 ee e8 a5 ff ff ff 85 c0 74 02 <0f> 0b 48 89 ee 48 89 df e8 27 10 00 00 85 c0 74 02 0f 0b 48 89 ee

and more

Is this normal?

**Comment by lwfinger** [comment]
No, that is not normal. The driver logged an error, and then printed the warning. Logging the error is sufficient. I have removed a could of warnings in this area.
